projects/batfish/src/test/java/org/batfish/grammar/cisco/CiscoGrammarTest.java

```diff
@@ -83,4 +90,62 @@ public void testBgpLocalAs() throws IOException {
             .getRemoteBgpNeighbor(),
         is(notNullValue()));
   }
+
+  @Test
+  public void testBgpRemovePrivateAs() throws IOException {
+    String testrigName = "bgp-remove-private-as";
+    String[] configurationNames = new String[] {"r1", "r2", "r3"};
+    Batfish batfish =
+        BatfishTestUtils.getBatfishFromTestrigResource(
+            TESTRIGS_PREFIX + testrigName, configurationNames, _folder);
+    SortedMap<String, Configuration> configurations = batfish.loadConfigurations();
+    Map<Ip, Set<String>> ipOwners = batfish.computeIpOwners(configurations, true);
+    batfish.initRemoteBgpNeighbors(configurations, ipOwners);
+    BdpDataPlanePlugin dataPlanePlugin = new BdpDataPlanePlugin();
+    dataPlanePlugin.initialize(batfish);
+    dataPlanePlugin.computeDataPlane(false);
+
+    // Check that 1.1.1.1/32 appears on r3
+    SortedMap<String, SortedMap<String, SortedSet<AbstractRoute>>> routes =
+        dataPlanePlugin.getRoutes();
+    SortedSet<AbstractRoute> r3Routes = routes.get("r3").get(Configuration.DEFAULT_VRF_NAME);
+    Set<Prefix> r3Prefixes = r3Routes.stream().map(r -> r.getNetwork()).collect(Collectors.toSet());
+    Prefix r1Loopback = new Prefix("1.1.1.1/32");
+    assertTrue(r3Prefixes.contains(r1Loopback));
+
+    // check that private AS is present in path in received 1.1.1.1/32 advert on r2
+    batfish.initBgpAdvertisements(configurations);
+    Configuration r2 = configurations.get("r2");
+    AtomicBoolean r2HasPrivate = new AtomicBoolean();
+    r2.getReceivedEbgpAdvertisements()
+        .stream()
+        .filter(a -> a.getNetwork().equals(r1Loopback))
+        .toArray(BgpAdvertisement[]::new)[0]
+        .getAsPath()
+        .getAsSets()
+        .stream()
+        .flatMap(asSet -> asSet.stream())
+        .forEach(
```

Comment:
can this and below be simplified using something like [Stream.anyMatch](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#anyMatch-java.util.function.Predicate-)?